regparser/tree/xml_parser/reg_text.py

```diff
@@ -177,9 +178,21 @@ def build_from_section(reg_part, section_xml):
                 n = Node(node_text[0], [], [m], source_xml=ch)
                 n.tagged_text = unicode(node_text[1])
                 nodes.append(n)
+
             if node_text[0].endswith('* * *'):
                 nodes.append(Node(label=[mtypes.INLINE_STARS]))
 
+    # special handling for reg J, whose definitions are inexplicably
+    # unnumbered and yet required to belong to node that precedes them
+    if reg_part == '1010' and nodes != [] and nodes[-1].text.find('As used in this part:') > -1:
+        for i, [plain_text, tagged_text] in enumerate(section_texts):
+            subnode = Node(plain_text, label=['b', str(i)], title="")
+            subnode.tagged_text = tagged_text
+            nodes[-1].children.append(subnode)
```

Comment:
I'd very, very, very strongly recommend against putting special cases like this in the main code base. It'll make maintaining this impossible and severely limits reuse. If you must make reg specific tweaks, add a hook of some sort.

That said, I don't think you need a hook here. Really, the reg authors don't view these nodes as "1010-2-b-1", "1010-2-b-2", etc. (or whatever label). They view them as "the definition of SomeTerm in 1010-2-b"; this is more or less how the paragraphs are referred to in later rules, and appears to be a common occurrence across regulations both within CFPB and without. What you want then, is something like 1010-2-b-SomeTerm for a label for this to work without manual intervention. That's a big undertaking though; in the meantime, perhaps it'd make sense to generalize away from 1010-2-b (or what have you). Consider something like "If you don't have a label, but you are a definition, incrementally number yourself". Look at the appendix parsing code -- it handles situations where there is no paragraph label more generally than the regtext and interpretation parsers.